pkg/ocmmanager/ocm_upgrade_config_manager.go

```diff
@@ -0,0 +1,439 @@
+	// HTTP client used for API queries (TODO: remove in favour of OCM SDK)
+	httpClient *http.Client
+}
+
+// Represents an unmarshalled Upgrade Policy response from Cluster Services
+type upgradePolicyList struct {
+	Kind  string          `json:"kind"`
+	Page  int64           `json:"page"`
+	Size  int64           `json:"size"`
+	Total int64           `json:"total"`
+	Items []upgradePolicy `json:"items"`
+}
+
+// Represents an unmarshalled individual Upgrade Policy response from Cluster Services
+type upgradePolicy struct {
+	Id                   string               `json:"id"`
+	Kind                 string               `json:"kind"`
+	Href                 string               `json:"href"`
+	Schedule             string               `json:"schedule"`
+	ScheduleType         string               `json:"schedule_type"`
+	UpgradeType          string               `json:"upgrade_type"`
+	Version              string               `json:"version"`
+	NextRun              string               `json:"next_run"`
+	PrevRun              string               `json:"prev_run"`
+	NodeDrainGracePeriod nodeDrainGracePeriod `json:"node_drain_grace_period"`
+	ClusterId            string               `json:"cluster_id"`
+}
+
+type nodeDrainGracePeriod struct {
+	Value int64  `json:"value"`
+	Unit  string `json:"unit"`
+}
+
+// Represents an unmarshalled Cluster List response from Cluster Services
+type clusterList struct {
+	Kind  string        `json:"kind"`
+	Page  int64         `json:"page"`
+	Size  int64         `json:"size"`
+	Total int64         `json:"total"`
+	Items []clusterInfo `json:"items"`
+}
+
+// Represents a partial unmarshalled Cluster response from Cluster Services
+type clusterInfo struct {
+	Id      string         `json:"id"`
+	Version clusterVersion `json:"version"`
+}
+
+type clusterVersion struct {
+	Id           string `json:"id"`
+	ChannelGroup string `json:"channel_group"`
+}
+
+type ocmRoundTripper struct {
+	authorization util.AccessToken
+}
+
+func (ort *ocmRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	authVal := fmt.Sprintf("AccessToken %s:%s", ort.authorization.ClusterId, ort.authorization.PullSecret)
+	req.Header.Add("Authorization", authVal)
+	transport := http.Transport{
+		TLSHandshakeTimeout: time.Second * 5,
+	}
+	return transport.RoundTrip(req)
+}
+
+// UpgradeConfigManager will trigger X every `WatchIntervalMinutes` and only stop if the operator is killed or a
+// message is sent on the stopCh
+func (s *osdUpgradeConfigManager) Start(stopCh <-chan struct{}) {
+	log.Info("Starting the upgradeConfigManager")
+
+	initialSync := true
+	for {
+
+		// Select a new watch interval with jitter
+		duration := durationWithJitter(s.config.GetWatchInterval(), JITTER_FACTOR)
+
+		// Don't wait if this is the first run since starting
+		if initialSync {
+			duration = 0
+			initialSync = false
+		}
+
+		select {
+		case <-time.After(duration):
+			_, err := s.RefreshUpgradeConfig()
```

Comment:
I'm thinking it might make sense to have some retry logic here. This will be a network request and susceptible to intermittent responses.
(If it makes sense) It can be a follow up card.

Reply:
and we probably need to alert if the endpoint is actually down (also follow up card if makes sense)

Reply:
I've created OSD-5413 and OSD-5414 for these